roxctl/generate/netpol/cta.go

```diff
@@ -1,28 +1,59 @@
+// Package netpol provides primitives for command 'roxctl generate netpol'
 package netpol
 
 import (
 	"os"
 	"path/filepath"
 	"strings"
 
-	"github.com/np-guard/cluster-topology-analyzer/pkg/controller"
+	npguard "github.com/np-guard/cluster-topology-analyzer/pkg/controller"
 	"github.com/pkg/errors"
 	"github.com/stackrox/rox/pkg/protoconv/networkpolicy"
 	v1 "k8s.io/api/networking/v1"
 )
 
+var errNPGErrorsIndicator = errors.New("there were errors during execution")
+
 func (cmd *generateNetpolCommand) generateNetpol() error {
-	recommendedNetpols, err := controller.PoliciesFromFolderPath(cmd.folderPath)
+	opts := []npguard.PoliciesSynthesizerOption{
+		npguard.WithLogger(newNpgLogger(cmd.env.Logger())),
+	}
+	if cmd.stopOnFirstError {
+		opts = append(opts, npguard.WithStopOnError())
+	}
+	npSynth := npguard.NewPoliciesSynthesizer(opts...)
```

Comment:
Maybe it would be worthwhile having this within the struct `generateNetpolCommand` and construct it with the options (i.e. adding the logger + stop on first error) within `generateNetpolCommand.construct()`.

This way you can inject a fake npguard synthesizer if we decide to add unit tests and it also follows the way we do depedency injection (i.e. service clients for gRPC APIs) in other commands.

Reply:
Done in [50f3f50](https://github.com/stackrox/stackrox/pull/2637/commits/50f3f508c0d3c5aef79f70788b521056da4feb41)